### PR TITLE
ISHEvent and ISHBackgroundTask field information is checked to add th…

### PR DESCRIPTION
**Pull request by ArianArt**
…em to theishTypeFieldDefinitions.

### CODE REVIEW

**Review by HildeVermeiren — APPROVED** [review]

**Review by OlegTokar — APPROVED** [review]

**Review by ddemeyer — CHANGES_REQUESTED** [review]

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs [hunk 1]
```diff
@@ -83,142 +87,101 @@ public IshTypeFieldSetup(ILogger logger, List<IshTypeFieldDefinition> ishTypeFie
...
-            ishTypeFieldDefinitions.Add(new IshTypeFieldDefinition(_logger, Enumerations.ISHType.ISHBackgroundTask, Enumerations.Level.History, false, false, true, false, false, false, true, true, false, "ENDDATE", Enumerations.DataType.DateTime, "", "The date time that this execution of the background task was finished"));
-            ishTypeFieldDefinitions.Add(new IshTypeFieldDefinition(_logger, Enumerations.ISHType.ISHBackgroundTask, Enumerations.Level.History, false, false, true, false, false, false, true, true, false, "HOSTNAME", Enumerations.DataType.String, "", "The host name from which the background task was created"));
+                //internal IshTypeFieldDefinition(ILogger logger, Enumerations.ISHType ishType, Enumerations.Level level,       bool isMandatory, bool isMultiValue, bool allowOnRead, bool allowOnCreate, bool allowOnUpdate, bool allowOnSearch, bool isSystem, bool isBasic, bool isDescriptive, string name, Enumerations.DataType dataType, string referenceLov, string description)
```
[the remark this answers is not in the document]
Comment (ArianArt) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs:154]: Yes, only layout..

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs [hunk 2]
```diff
@@ -83,142 +87,101 @@ public IshTypeFieldSetup(ILogger logger, List<IshTypeFieldDefinition> ishTypeFie
...
-            ishTypeFieldDefinitions.Add(new IshTypeFieldDefinition(_logger, Enumerations.ISHType.ISHBackgroundTask, Enumerations.Level.History, false, false, true, false, false, false, false, true, true, false, "STARTDATE", Enumerations.DataType.DateTime, "", "", "The date time that this execution of the background task was started"));
-            ishTypeFieldDefinitions.Add(new IshTypeFieldDefinition(_logger, Enumerations.ISHType.ISHBackgroundTask, Enumerations.Level.History, false, false, true, false, false, false, false, true, true, false, "ENDDATE", Enumerations.DataType.DateTime, "", "", "The date time that this execution of the background task was finished"));
-            ishTypeFieldDefinitions.Add(new IshTypeFieldDefinition(_logger, Enumerations.ISHType.ISHBackgroundTask, Enumerations.Level.History, false, false, true, false, false, false, false, true, true, false, "HOSTNAME", Enumerations.DataType.String, "", "", "The host name from which the background task was created"));
+            var ishTypeFieldDefinitions = new List<IshTypeFieldDefinition>
```
[the remark this answers is not in the document]
Comment (ArianArt) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs:117]: The layout is changed to resolve the warning.

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs [hunk 3]
```diff
@@ -71,9 +74,10 @@ public IshTypeFieldSetup(ILogger logger, string xmlIshFieldSetup)
                 }
             }
 
-            AddIshBackgroundTaskTableFieldSetup();
-            AddIshEventTableFieldSetup();
-            AddIshTranslationJobItemTableFieldSetup();
+            if (!hasIshBackgroundTaskFieldDefinitions)
+            { AddIshBackgroundTaskTableFieldSetup(); }
+            if (!hasIshEventFieldDefinitions)
+            { AddIshEventTableFieldSetup(); }
```
[the remark this answers is not in the document]
Comment (HildeVermeiren) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs:76]: Previously we had an empty method AddIshTranslationJobItemTableFieldSetup which is not used...so do you really want to keep this empty method and put a check around it?

**Review by HildeVermeiren — APPROVED** [review]

**Review by OlegTokar — APPROVED** [review]

File: Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ddemeyer) [reply Source/ISHRemote/Trisoft.ISHRemote/Objects/IshTypeFieldSetup.cs:76]: Although commented out, it was still a building block on route to #69 ... preferably the RetrieveTypeFieldSetup API call offers these two object types as well.

**Review by ddemeyer — APPROVED** [review]

**Pull request merged**